sources/OpenMcdf.Extensions/OLEProperties/OLEPropertiesContainer.cs

```diff
@@ -154,9 +154,10 @@ internal OLEPropertiesContainer(CFStream cfStream)
                     UserDefinedProperties.properties.Add(op);
                 }
 
-                UserDefinedProperties.PropertyNames = (Dictionary<uint, string>)pStream.PropertySet1.Properties
+                var existingPropertyNames = (Dictionary<uint, string>)pStream.PropertySet1.Properties
                     .Where(p => p.PropertyType == PropertyType.DictionaryProperty).FirstOrDefault()?.Value;
 
+                UserDefinedProperties.PropertyNames = existingPropertyNames ?? new Dictionary<uint, string> ();
```

Comment:
The test file 'english.presets.doc' contains a UserDefinedProperties section, but it doesn't contain a dictionary property, which as it stands prevents a user from adding new properties to the existing section.

The Microsoft documentation at https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-oshared/1414f765-ea8e-4a1e-a19c-641500e5f93d says that the UserDefinedProperty section MUST have a dictionary so I'm not sure why this file doesn't have one, but - If it must be there then i think it's reasonable to always have ```PropertyNames``` set to a not-null value for the UserDefined property section?